pgtoolkit/hba.py

```diff
@@ -455,13 +455,16 @@ def merge(self, other: "HBA") -> bool:
         return lines != self.lines
 
 
-def parse(file: Union[str, Iterable[str]]) -> HBA:
+def parse(file: Union[str, Iterable[str], Path]) -> HBA:
     """Parse a `pg_hba.conf` file.
 
-    :param file: Either a line iterator such as a file-like object or a string
+    :param file: Either a line iterator such as a file-like object, a path or a string
         corresponding to the path to the file to open and parse.
     :rtype: :class:`HBA`.
     """
+    if isinstance(file, Path):
+        file = str(file)
+
     if isinstance(file, str):
```

Comment:
I think `isinstance(file, (str, Path)):` is sufficient as `open()` can handle a `Path` value, so no need to convert the `Path` into a string.